resources/js/view-manager.js

```diff
@@ -0,0 +1,59 @@
+const viewManager = () => ({
+    viewedPosts: [],
+    addViewedPost(postId) {
+        this.viewedPosts = [...this.viewedPosts, postId];
+    },
+    sendViewedPosts() {
+        let data = JSON.parse(localStorage.getItem('viewedPosts')) || [];
+        let previousViewedPosts = data.filter(function (post) {
+            if (post === null) {
+                return false;
+            }
+            const twoHours = 2 * 60 * 60 * 1000;
+            return new Date().getTime() - post.dateTime < twoHours;
+        });
+        let previousViewedPostIds = previousViewedPosts.map(post => post.postId);
+        let viewedPosts = this.viewedPosts.filter(postId => !previousViewedPostIds.includes(postId));
+        this.viewedPosts = [];
+        if (viewedPosts.length > 0) {
+            this.$wire.call('updateViews', viewedPosts);
+            viewedPosts = viewedPosts.map(function (postId) {
+                return {
+                    postId: postId,
+                    dateTime: new Date().getTime()
+                };
+            });
+            viewedPosts = [...previousViewedPosts, ...viewedPosts];
+            localStorage.setItem('viewedPosts', JSON.stringify(viewedPosts));
```

Comment:
We will only store newly viewed questions and for other questions server will handle them. so mostly if this kind of thing happens then newly viewed can be sent twice to the server.

btw it will always have posts less viewed within two hours as for every server call we remove older posts.